### DISCUSSION TITLE
Should tabset work in a list?

### DISCUSSION THREAD

**Discussion by cudawarped** [quarto-dev/quarto-cli#5464] [Q&A] [answered]
I am trying to use [tabset](https://quarto.org/docs/output-formats/html-basics.html#tabsets) while in a list without much luck.

If I try to indent the tabset then the tabset is not rendered and if I don't but then try to continue a list as below

````
1.

::: {.panel-tabset}
  ## R

  ``` {.r}
  fizz_buzz <- function(fbnums = 1:50) {
  ```

  ## Python

  ``` {.python}
  def fizz_buzz(num):
  ```

:::

    continue list
````

Then the `continue list` text is surrounded by the `<pre><code>` tags.

Is this a bug or am I trying something (tabset inside a list) which is not reasonable?

**Comment by mcanouil** [comment, marked as answer]
I don't know if it's the way your formatted your post or your actual document, but it fails to follow Pandoc/Markdown rules for list.

<details><summary>See the below properly formatted document.</summary>

````qmd
---
title: "Demo"
format: html
---

1. Item 1

   ::: {.panel-tabset}
   ## R code

   ```r
   1 + 1 # R
   ```

   ## Python code

   ```python
   1 + 1 # Python
   ```
   :::

2. Item 2
  
   ::: {.panel-tabset}
   ## R code

   ```r
   2 + 2 # R
   ```

   ## Python code

   ```python
   2 + 2 # Python
   ```
   :::

Something that is not an item.
````

</details>
<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/34dfdec8-57ba-4cb5-b4b9-ab8a36ae6114">

**Reply by cudawarped** [reply]
Thank you thats great.  So the key is to have the same number of spaces before ::: as there are preceding Item 1?

**Reply by cscheid** [reply]
They need to be indented just like the content of the list item.

**Reply by cudawarped** [reply]
Fantastic thank you,.
